client/src/App.js

```diff
@@ -1,28 +1,52 @@
-import React from 'react';
-import { createTheme, ThemeProvider } from '@material-ui/core';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { ThemeProvider } from '@material-ui/core';
 import Header from './components/Header';
 import Footer from './components/Footer';
 import './App.scss';
-
-const theme = createTheme({
-  palette: {
-    primary: {
-      main: '#66737f',
-      light: '#c4c4c4',
-    },
-    secondary: {
-      main: '#eeb955',
-    },
-  },
-  typography: {
-    fontFamily: 'Lato',
-    button: {
-      textTransform: 'none',
-    },
-  },
-});
+import { userActions } from './store/user/userSlice';
+import requestApi from './helpers/requestApi';
+import theme from './theme';
 
 function App() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    localStorage.setItem(
+      'token',
+      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImthdHJpbkBnbWFpbC5jb20iLCJ1c2VySWQiOiI2MTcwNGFjMjBjMjViYTRmOGYyZDNjYzQiLCJpYXQiOjE2MzcyMzE4NzUsImV4cCI6MTYzNzIzNTQ3NX0.vodzIflLkypHM67_xwgZxzEbH-HE_Cu-4ua19PJx-hk'
+    );
+    const fetchUserData = async () => {
```

Comment:
you should do it in a middleware, if you're going to use redux-thunks
your component should trigger the request only, it shouldn't contain a request logic